src/microbe_stage/gui/PatchExtinctionBox.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using Godot;
+
+public class PatchExtinctionBox : Control
+{
+    [Export]
+    public NodePath PatchMapDrawerPath = null!;
+
+    [Export]
+    public NodePath PatchDetailsPanelPath = null!;
+
+    [Export]
+    public NodePath AnimationPlayer = null!;
+
+    private PatchMapDrawer mapDrawer = null!;
+    private PatchDetailsPanel detailsPanel = null!;
+    private AnimationPlayer animationPlayer = null!;
+
+    public PatchMap? Map
+    {
+        get => mapDrawer.Map;
+        set
+        {
+            mapDrawer.Map = value;
+            mapDrawer.SetPatchEnabledStatuses(value!.Patches.Values, p => p.GetSpeciesPopulation(PlayerSpecies) > 0);
```

Comment:
This should throw an explicit exception if value is null, if this doesn't work with null values